SmartDeviceLink/SDLIAPSession.m

```diff
@@ -139,6 +273,51 @@ - (SDLStreamErrorHandler)streamErroredHandler {
     };
 }
 
+- (SDLStreamHasSpaceHandler)streamHasSpaceHandler {
+    __weak typeof(self) weakSelf = self;
+    
+    return ^(NSStream *stream) {
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        
+        if (strongSelf.isDataSession){
+            SessionSendHandler sendCompletion;
+            @synchronized (strongSelf) {
+                sendCompletion = strongSelf.sendBlock;
+            }
+            
+            if (sendCompletion != nil){
+                NSError *err = nil;
+                BOOL sendComplete = sendCompletion(&err);
+                if (err != nil){
+                    NSLog(@"ERROR!!! Output stream reported %@", err);
```

Comment:
I have a somewhat related question about the ~9s delay after the accessory connect event before the control session establishment is attempted. In my testing of the background thread changes, the various crashes I've seen due to the cross-thread access to the output stream are no longer reproducible. 

However, the 9s delay causes the app to resume much more slowly than one would ideally like. There is also an issue with running timers in the background that causes the app to not resume if the control session is not established right away. If the timer is needed and if it is necessary to retry more than once, the code should be placed in a background task when the app is backgrounded. I have another PR in the hopper for the background task, but before I submit that, could we eliminate the timer?